src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.Scenario;
+import executor.service.service.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/**
+ * The {@code ParallelFlowExecutorServiceImpl} class implements the {@link ParallelFlowExecutorService} interface
+ * to execute ScenarioSourceListener, ProxySourcesClient, and ExecutionService in parallel multithreading mode.
+ * It manages the execution of these services by utilizing an {@link ExecutorService} and {@link TasksFactory}.
+ * This class also provides a method to gracefully shut down the executor service.
+ * <p>
+ *
+ * @author Oleksandr Tuleninov
+ * @version 01
+ * @see executor.service.service.ExecutionService
+ * @see executor.service.service.ScenarioSourceListener
+ * @see executor.service.service.ProxySourcesClient
+ */
+public class ParallelFlowExecutorServiceImpl implements ParallelFlowExecutorService {
+
+    private static final Logger log = LoggerFactory.getLogger(ParallelFlowExecutorServiceImpl.class);
+    private static boolean FLAG = true;
+
+    private final ExecutorService threadPoolExecutor;
+    private final ExecutionService service;
+    private final ScenarioSourceListener scenarioSourceListener;
+    private final ProxySourcesClient proxySourcesClient;
+    private final TasksFactory tasksFactory;
+
+    public ParallelFlowExecutorServiceImpl(ExecutorService threadPoolExecutor,
+                                           ExecutionService service,
+                                           ScenarioSourceListener scenarioSourceListener,
+                                           ProxySourcesClient proxySourcesClient,
+                                           TasksFactory tasksFactory) {
+        this.threadPoolExecutor = threadPoolExecutor;
+        this.service = service;
+        this.scenarioSourceListener = scenarioSourceListener;
+        this.proxySourcesClient = proxySourcesClient;
+        this.tasksFactory = tasksFactory;
+    }
+
+    /**
+     * Starts the ScenarioSourceListener, ProxySourcesClient, and ExecutionService
+     * in parallel multithreading mode.
+     */
+    @Override
+    public void execute() {
+        Future<ItemQueue> futureScenarios
+                = threadPoolExecutor.submit(tasksFactory.createTaskWorker(scenarioSourceListener));
+
+        Future<ItemQueue> futureProxies
+                = threadPoolExecutor.submit(tasksFactory.createTaskWorker(proxySourcesClient));
+
+        executeScenarioAndProxy(futureScenarios, futureProxies);
+    }
+
+    /**
+     * Initiates an orderly shutdown in which previously submitted tasks are executed,
+     * but no new tasks will be accepted.
+     */
+    @Override
+    public void shutdown() {
+        FLAG = false;
+        threadPoolExecutor.shutdown();
+    }
```

Comment:
Witch service use this method?

Reply:
At the lecture, the mentors said that this method has a place. Therefore, I suggest leaving it. But if it is important to you, we can consider this issue again at the rally. Please write.

I would like to note that threadPoolExecutor implements ExecutorService, and they have this method. Confirmation below:
![image](https://github.com/GFLCourses5/worker/assets/140727960/7dfbc39f-b59f-4914-a458-2fefd54b746c)
![image](https://github.com/GFLCourses5/worker/assets/140727960/e146d5a2-4021-4bb7-aa34-4b421dbd4b9a)
Also, this method might be useful when we switch to Spring to properly terminate threadPoolExecutor,
this is possible when using a separate Endpoint controller.